treewide: replace CRLF with LF

seragh opened the pull request.

hnyman commented:
This looks suspicious on the part where you modify patches that modify existing files from upstream repos. 

Upstream repos may contain CRLFs (possibly erroneously, but still) and the patches will then need to contain CRLFs, too.

Recent example:
http://buildbot.openwrt.org:8010/broken_packages/ar71xx/lua-openssl/compile.txt
```
Applying ./patches/0001-Revise-Makefile.patch using plaintext: 
patching file Makefile
Hunk #1 FAILED at 1 (different line endings).
1 out of 1 hunk FAILED -- saving rejects to file Makefile.rej
```
That was fixed yesterday with https://github.com/openwrt/packages/pull/3415 by adding CRLFs to the patch to make it to match the upstream source code...

(CRLFs are not visible in Github , but here you can see them: See https://git.lede-project.org/?p=feed/packages.git;a=commitdiff;h=d0697b488b6145bd07efef6e8be889bc43673145;hp=fba5959f4e7d77ab75d24f5517e71847f0fdccb9

seragh committed:
treewide: replace CRLF with LF

Ignoring patches having CRLF, as conversion might break applying them to
upstream sources out of our control.

Signed-off-by: Ralph Sennhauser <ralph.sennhauser@gmail.com>

seragh commented:
On Mon, 31 Oct 2016 00:52:53 -0700
Hannu Nyman <notifications@github.com> wrote:

> This looks suspicious on the part where you modify patches that
> modify existing files from upstream repos. 
> 
> Upstream repos may contain CRLFs (possibly erroneously, but still)
> and the patches will then need to contain CRLFs, too.
> 
> Recent example:
> http://buildbot.openwrt.org:8010/broken_packages/ar71xx/lua-openssl/compile.txt
> ```
> Applying ./patches/0001-Revise-Makefile.patch using plaintext: 
> patching file Makefile
> Hunk #1 FAILED at 1 (different line endings).
> 1 out of 1 hunk FAILED -- saving rejects to file Makefile.rej
> ```
> That was fixed yesterday with
> https://github.com/openwrt/packages/pull/3415 by adding CRLFs to the
> patch to make it to match the upstream source code...
> 
> (CRLFs are not visible in Github , but here you can see them: See
> https://git.lede-project.org/?p=feed/packages.git;a=commitdiff;h=d0697b488b6145bd07efef6e8be889bc43673145;hp=fba5959f4e7d77ab75d24f5517e71847f0fdccb9
> 
> 
> 
Hi Hannu,

Indeed, "patch" only handles the case where the file use LF but the
patch uses CRLF. The other way around it complains about wrong line
endings.

Good catch, pushed an update

Thanks
Ralph

seragh commented:
As part of a series to fix cryptodev support in openssl quite a bunch of the files converted will be removed anyway.

So retracting this request for now, and then when not in a hurry test if those pesky patches indeed need CRLF. This asymmetry in "patch" behaviour wrt to line endings  sure rains on the parade.
